src/EditorFeatures/CSharpTest/CodeRefactorings/AddMissingImports/AddMissingImportsRefactoringProviderTests.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editor.UnitTests;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.PasteTracking;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.VisualStudio.Composition;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.AddMissingImports
+{
+    [UseExportProvider]
+    [Trait(Traits.Feature, Traits.Features.AddMissingImports)]
+    public class AddMissingImportsRefactoringProviderTests : AbstractCSharpCodeActionTest
+    {
+        protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
+        {
+            var testWorkspace = (TestWorkspace)workspace;
+            var pasteTrackingService = testWorkspace.ExportProvider.GetExportedValue<PasteTrackingService>();
+            return new AddMissingImportsRefactoringProvider(pasteTrackingService);
+        }
+
+        protected override TestWorkspace CreateWorkspaceFromFile(string initialMarkup, TestParameters parameters)
+        {
+            var workspace = TestWorkspace.CreateCSharp(initialMarkup, exportProvider: GetExportProvider());
+
+            var diagnosticAnalyzerService = (DiagnosticAnalyzerService)workspace.ExportProvider.GetExportedValue<IDiagnosticAnalyzerService>();
+            diagnosticAnalyzerService.CreateIncrementalAnalyzer(workspace);
+
+            // Treat the span being tested as the pasted span
+            var hostDocument = workspace.Documents.First();
+            var pastedTextSpan = hostDocument.SelectedSpans.FirstOrDefault();
+
+            if (!pastedTextSpan.IsEmpty)
+            {
+                var pasteTrackingService = workspace.ExportProvider.GetExportedValue<PasteTrackingService>();
+                pasteTrackingService.RegisterPastedTextSpan(hostDocument.GetTextView(), hostDocument.TextBuffer, pastedTextSpan);
```

Comment:
we should have tests taht work by actually sending in the paste command.  See the CompletionCommandHandlerTests for examples on how to do this sort of thing.